test/FIFOMarket.test.ts

```diff
@@ -9,46 +11,99 @@ import {
   createRemovalTokenId,
 } from '@/test/helpers';
 
+interface RemovalDataForListing {
+  amount: number;
+  vintage?: number;
+  supplier?: string;
+}
+
+interface RemovalDataFromListing {
+  listedRemovalIds: BigNumber[];
+  totalAmountOfSupply: number;
+  totalAmountOfSuppliers: number;
+  totalAmountOfRemovals: number;
+}
+
+const parseNumberToBigNumber = (numberToParse: number): string =>
+  hre.ethers.utils.parseUnits(numberToParse.toString()).toString();
+
+const getTotalAmountOfSupply = (removals: RemovalDataForListing[]): number =>
+  removals.reduce((sum, removal) => sum + removal.amount, 0);
+
+const getTotalAmountOfSuppliers = (removals: RemovalDataForListing[]): number =>
+  removals.reduce(
+    (supplierSet, removal) => supplierSet.add(removal.supplier),
+    new Set()
+  ).size ?? 1;
+
+const getTotalAmountOfRemovals = (removals: RemovalDataForListing[]): number =>
```

Comment:
this seems oddly gratuitous :joy:

Reply:
Now that's what I call functional programming!